base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -233,12 +222,44 @@ impl LMDBDatabase {
         Ok(())
     }
 
-    #[allow(clippy::ptr_arg)]
+    fn append_header_accumulated_data(
+        &mut self,
+        txn: &WriteTransaction<'_>,
+        header: &BlockHeader,
+    ) -> Result<(), ChainStorageError>
+    {
+        debug!(
+            target: LOG_TARGET,
+            "height={} ({}), prev={}",
+            header.height,
+            header.hash().to_hex(),
+            header.prev_hash.to_hex()
+        );
+        let prev_data = if header.height == 0 {
+            BlockHeaderAccumulatedData::default()
```

Comment:
This struct should not have a default, it should always have a specific value attached to a header.
```suggestion
            BlockHeaderAccumulatedData::new(header.hash(), total_kernel_offset)
```

Reply:
This is the `new` functions signature
```rust
 pub fn new(
        kernels: PrunedHashSet,
        outputs: PrunedHashSet,
        range_proofs: PrunedHashSet,
        deleted: Bitmap,
        total_kernel_sum: Commitment,
        total_utxo_sum: Commitment,
    ) -> Self
```

A default would populate the struct with zero/empty values. I stand corrected but I think that is intuitive to the Default trait. Perhaps `BlockAccumulatedData::empty()` is more descriptive.